src/lib/validation/form/deprecated-appearance.js

```diff
@@ -0,0 +1,87 @@
+const semver = require('semver');
+const { getNodes } = require('../../forms-utils');
+
+// Based on https://forum.getodk.org/t/spec-proposal-harmonize-compact-compact-n-horizontal-horizontal-compact-appearances/15565/32
+// Note the order of evaluation is important
+const getDeprecatedAppearances = (apiVersion) => [
+  {
+    match: appearance => appearance.match(/(?:^|\s)horizontal-compact(?:$|\s)/),
+    replacement: () => 'columns-pack',
```

Comment:
My old-school-self is reading this as `getReplacement` 😄 I like the practically of 1 word

Reply:
Well... now that you mention it, I think that started with this just being a field containing a string and eventually refactored it to be a function (but forgot to rename it)...  Now I have to go and change this name or it will drive me crazy!  :laughing: